test/bpf/verifier-test.sh

```diff
@@ -289,9 +289,14 @@ function handle_developers {
 	fi
 }
 
+function mount_bpf {
+	mount | grep -q 'type bpf' || mount -t bpf bpf /sys/fs/bpf
+}
+
 function main {
 	handle_args "$@"
 	handle_developers
+	mount_bpf
```

Comment:
Why is this happening for Kind but not when running our usual Jenkins jobs?

It feels like an odd place to mount the bpffs. Shouldn't it be mounted as part of the Kind provisioning?

Reply:
My guess is that the regular Jenkins jobs are guaranteed to install Cilium and so can rely on Cilium to set up bpffs for us. Whereas when I try this locally, I deliberately skip provisioning and jump straight to the K8sVerifier test.

I vaguely recall that for some reason kind by default ends up with sysfs mounted at /sys/fs/bpf 🤷 

I'm open to ideas on how to mount it as part of kind provisioning, but I'm not aware of any ways to make such configuration. The configurations I added in these commits are just mounting specific paths into the kind containers from the host, but we can't use this approach for bpffs or we may end up causing different nodes to reuse the same underlying FS and interfere with each other. In theory I guess the setup script could somehow go into the individual kind nodes and mount bpffs, not sure quite how to do this in a simple way (other than maybe deploying some dummy daemonset to mount the bpffs or something...?) . In the end it seemed simpler to just have a "check and mount bpffs" step in this script, since we know it will run where we need it to run and with the appropriate privileges, and this is a prerequisite to run the test.

Reply:
> My guess is that the regular Jenkins jobs are guaranteed to install Cilium and so can rely on Cilium to set up bpffs for us. Whereas when I try this locally, I deliberately skip provisioning and jump straight to the K8sVerifier test.

AFAIK, if K8sVerifier runs first, Cilium is not installed, as K8sVerifier doesn't install it itself. Hence why I'm surprised :slightly_smiling_face: 

Could we mount it as part of Verifier.go? We already deploy the test pod there, so it might make sense to also prepare the node (i.e., mount bpffs) in that same function.

Reply:
That works too. I changed the patch to follow this suggestion, PTAL.